pkg/cache/multi_namespace_cache.go

```diff
@@ -207,11 +332,18 @@ func (c *multiNamespaceCache) Get(ctx context.Context, key client.ObjectKey, obj
 	}
 
 	if !isNamespaced {
+		if c.clusterCache == nil {
+			return fmt.Errorf("unable to get: %v because cluster-scoped cache does not exist", key)
+		}
 		// Look into the global cache to fetch the object
 		return c.clusterCache.Get(ctx, key, obj)
 	}
 
 	cache, ok := c.namespaceToCache[key.Namespace]
+	if !ok {
+		// Use the default/catch-all namespace cache if we have one.
+		cache, ok = c.namespaceToCache[corev1.NamespaceAll]
+	}
 	if !ok {
 		return fmt.Errorf("unable to get: %v because of unknown namespace for the cache", key)
```

Comment:
Should we rephrase this error a bit? (same in l.378)

E.g. 
```go
			return fmt.Errorf("unable to get: %v because neither a per-namespace nor a default namespaced cache exists", listOpts.Namespace)
```
Similar in l.378 just that it should be "list" there